src/js/components/modals/ServiceFormModal.js

```diff
@@ -269,6 +269,35 @@ class ServiceFormModal extends mixin(StoreMixin) {
   shouldDisableForm(service) {
     let containerSettings = service.getContainerSettings();
 
+    let portDefinitions = service.getPortDefinitions();
+    let name = new RegExp(`${service.getId()}`, 'g');
+
+    if (portDefinitions) {
+      if (!portDefinitions.some(function (port) {
+        if (port.labels == null || Object.keys(port.labels).length === 0) {
+          return true;
+        }
+        return Object.keys(port.labels).some(function (key) {
+          return name.test(port.labels[key]);
+        });
+      })) {
+        return true;
+      }
+    }
+
+    if (containerSettings && containerSettings.docker && containerSettings.docker.portMappings) {
+      if (!containerSettings.docker.portMappings.some(function (port) {
+        if (port.labels == null || Object.keys(port.labels).length === 0) {
+          return true;
+        }
+        return Object.keys(port.labels).some(function (key) {
+          return name.test(port.labels[key]);
```

Comment:
Changed the regexp to `let name = new RegExp(`^${service.getId()}:[0-9]*$`, 'g');` 
But we have to check against `${service.getId()}:${port.port || port.containerPort || port.hostPort}` since the port can be in a lot of fields :-(.